lib/net/pool.js

```diff
@@ -2139,7 +2347,8 @@ class Pool extends EventEmitter {
   async _handleHeaders(peer, packet) {
     const headers = packet.items;
 
-    if (!this.checkpoints)
+    if (!this.checkpoints && !this.options.neutrino)
```

Comment:
lets talk about how `_handleHeaders()` should look. I think it should look like handleBlock essentially which makes the most sense, in which we just pass the headers to chain to verify and store. However FullNode has this headerchain thing in pool which should I think we should move this logic to a new function and call that from here in full mode. (and sigh yes in spv mode too just to not break anything, but in the future we can make that a lot smarter too)

Reply:
For example in the pool-only headers thing, we verify the header PoW and prevblock right here, but all that should be done by chain in neutrino mode